api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -734,9 +734,11 @@ public User getUserByEmail(String email) throws APIException {
 	@Transactional(readOnly = true)
 	public User getUserByActivationKey(String activationKey) {
 		LoginCredential loginCred = dao.getLoginCredentialByActivationKey(activationKey);
-		String[] credTokens = loginCred.getActivationKey().split(":");
-		if(loginCred != null  && (System.currentTimeMillis() <= Long.parseLong(credTokens[1]) )) {
-			return getUser(loginCred.getUserId());
+		if (loginCred != null) {
+			String[] credTokens = loginCred.getActivationKey().split(":");
+			if (System.currentTimeMillis() <= Long.parseLong(credTokens[1])) {
+				return getUser(loginCred.getUserId());
```

Comment:
I'd actually like to see a very specific exception getting thrown if the credential has expired so that the calling code know what to do next